types/underscore/index.d.ts

```diff
@@ -116,27 +119,16 @@ declare module _ {
         I extends undefined ? T // default iteratee is _.identity
         : I extends (...args: any[]) => infer R ? R
         : I extends keyof T ? T[I]
-        : I extends EnumerableKey | Array<EnumerableKey> ? any
+        : I extends EnumerableKey |EnumerableKey[] ? any
         : I extends Partial<T> ? boolean
         : never;
 
     type PropertyTypeOrAny<T, K> = K extends keyof T ? T[K] : any;
 
-    type IterateePropertyShorthand = string | number;
-
-    interface MemoIterator<T, TResult, V = List<T>> {
-        (prev: TResult, curr: T, index: number, list: V): TResult;
-    }
-
-    interface MemoObjectIterator<T, TResult, V = Dictionary<T>> {
```

Comment:
How do you feel about these going away? Not sure if it's worth keeping them for backwards compatibility or not.

Reply:
Never mind, per the CI failure [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46120/checks?check_run_id=880589818) Backbone references these right now so I need to keep them around for the moment.

I guess I should probably also update Backbone mixin types once this is all done.

Reply:
Looking at those types, it looks like you're an owner of those as well, which should make that easier 😄

Reply:
Hmm, in the interest of being DRY though, I should probably redeclare these as aliases to `CollectionIterator` and `MemoCollectionIterator`.